### PR TITLE
Explore: Enable resize of split pane

### LINKED ISSUE #58582: Have Explore use generic splitpane wrapper

### PR DESCRIPTION
**Pull request by gelicia**
**What is this feature?**

This implements split pane resize in Explore using the generic split component we implemented in #58380.

This contains the features added in #54420 (this was later reverted). These changes were manually added back, so some discrepancies might exist.

**Which issue(s) does this PR fix?**:
Fixes #58582

### CODE REVIEW

**Comment by Elfo404** [comment]
didn't look at the code yet, but i was testing this out and this behaviour was kinda weird: i'm able to resize the a pane down to 0px and only when releasing the mouse button it'd get resized to its minimum size. that feels kinda odd and not consistent with what happens in dashboards
https://user-images.githubusercontent.com/1170767/202148915-875e8ad4-cf1b-4350-9a60-b8dcc812250d.mov

**Comment by ifrost** [comment]
The logs panel seems very laggy when being resized. It's also happening in the panel editor when a logs panel is used. Do you think it'd make sense to add some debounce to the resize? 🤔 

https://user-images.githubusercontent.com/745532/202159810-6a15eeff-5b8e-4a60-9818-cd66e3e1bb7e.mov

File: public/app/features/explore/state/main.ts [hunk 1]
```diff
@@ -179,6 +192,38 @@ export const exploreReducer = (state = initialExploreState, action: AnyAction):
     return {
       ...state,
       ...targetSplit,
+      largerExploreId: undefined,
```
[the remark this answers is not in the document]
Comment (gelicia) [reply public/app/features/explore/state/main.ts:195]: Let's make a refactoring ticket about this and we can discuss it then.

**Comment by gelicia** [comment]
> i'm able to resize the a pane down to 0px and only when releasing the mouse button it'd get resized to its minimum size. that feels kinda odd and not consistent with what happens in dashboards

this is now using library params that should make it behave consistent with dashboards

**Comment by ifrost** [comment]
> The logs panel seems very laggy when being resized. It's also happening in the panel editor when a logs panel is used. Do you think it'd make sense to add some debounce to the resize?

This is still happening (at least on my laptop). Definitely, something with logs panel performance, happens only if logs are loaded. See video below (first part is without logs, second with logs):

https://user-images.githubusercontent.com/745532/202421813-ebb2cbba-e814-42de-8c73-aa959f97db73.mov

**Review by ifrost — APPROVED** [review]
Nice split 👍

**Pull request merged**